.github/workflows/pr-test.yml

```diff
@@ -494,3 +494,61 @@ jobs:
         run: |
           pipenv run pytest
         working-directory: ${{ env.WORKING_DIRECTORY }}
+
+  eval-keys-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11  # v3
+
+      - name: Get changed Python files
+        id: changed-files
+        uses: tj-actions/changed-files@6b2903bdce6310cfbddd87c418f253cf29b2dec9 # v44
+        with:
+          files: checkov/**/*.py
+
+      - name: Validate 'BaseResourceCheck' use contains eval keys
+        if: steps.changed-files.outputs.any_changed == 'true'
+        run: |
+          # Define an array of exceptions (files to skip)
+          EXCEPTIONS=(
+            "base_resource_check.py" 
+            "VPCDefaultNetwork.py"
+            "IAMUserNotUsedForAccess.py"  # Whole Resource type check
+          )
+          
+          echo "Changed files:"
+          echo "${{ steps.changed-files.outputs.all_changed_files }}"
+          
+          EXIT_CODE=0
+          IFS=$'\n'  # Change Internal Field Separator to handle spaces in filenames too
+          for file in $(echo "${{ steps.changed-files.outputs.all_changed_files }}" | tr ',' '\n'); do
+            # Check if the file is in the list of exceptions
+            SKIP_FILE="false"
+            for exception in "${EXCEPTIONS[@]}"; do
+              # If the file ends with one of the exception file names, skip it
+              if [[ "$file" == *"$exception" ]]; then
+                echo "Skipping $file (allowed exception)"
+                SKIP_FILE="true"
+                break
+              fi
+            done
+          
+            # Only run checks if not in exceptions list
+            if [[ "$SKIP_FILE" == "false" ]]; then
+              # If file contains 'BaseResourceCheck', check for 'get_inspected_key' or 'evaluated_keys'
```

Comment:
this is only a partial validation, as `BaseResourceCheck` doesn't exist in all of the code.
for example, if we inherited from it a class called `BaseCloudformationResourceCheck`, any check which inerits from this one will not catch this.
I don't mind adding this check for now, but a better solution would be a python linter (flake8-plugin for example) which can catch this issue